### PR TITLE
Fixes #217 and #218 adds support for autoapi

### LINKED ISSUE #217: Integrate javasphinx with yaydoc
### This is a
- [ ] Bug report
- [x] Feature request

### Description

Parent Issue - #203 

Integrate javasphinx so that java APIs can be documented

### LINKED ISSUE #218: Add support for installing with setup.py and mocking packages for sphinx-apidoc
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

Currently If we a requirements.txt is found in the target repo's root, we pip install it.

### Expected behaviour

- If requirements.txt is not present but a setup.py is, we should install with setup.py
- Some projects may depend on C- extensions which cannot be installed with pip, for such cases allow users to specify packages to mock.

### PR DESCRIPTION
**Pull request by pri22296**
## Description

Variables introduced

* `AUTOAPI_PYTHON`
* `AUTOAPI_PYTHON_PATH`
* `AUTOAPI_JAVA`
* `AUTOAPI_JAVA_PATH`

Variables removed

* `PYTHON_PACKAGE`

`PYTHON_PACKAGE` was no longer needed. Now we automatically search for all packages in `AUTOAPI_PYTHON_PATH`

To document APIs, we walk the repository looking for source files and extract documentation from comments in sourcecode. To extract Python docstring, we use sphinx-apidoc. To extract Java docstrings, we use [javasphinx](http://bronto-javasphinx.readthedocs.io/en/latest/)

## Related Issue

Fixes #217
Fixes #218 

## Motivation and Context

Adds support for generating API documentation by walking the repository and extracting any docstrings from source files.

## How Has This Been Tested?

Travis build log: https://travis-ci.org/pri22296/testYaydoc2/builds/249889732
Test Repository: https://github.com/pri22296/testYaydoc2
Generated Javadoc: https://pri22296.github.io/testYaydoc2/source/packages.html

### Generated TOCTREE for javadocs

screenshot 171

### Generated API doc for a sample class

screenshot 172

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by pri22296** [comment]
@imujjwal96 @sch00lb0y Please review

**Comment by imujjwal96** [comment]
What's the use case? 
How does it work?

**Comment by pri22296** [comment]
@imujjwal96 I have updated the description. Basically I added https://github.com/pri22296/testYaydoc2/tree/master/src directory in my test repo. javasphinx generated the javadocs from it automatically. You can see one of the generated files [here](https://github.com/pri22296/testYaydoc2/blob/gh-pages/_sources/source/ai/susi/EmailHandler.rst.txt).  Users will be able to override the generated source files by providing their own source files.

File: ghpages_deploy.sh [hunk 1]
```diff
@@ -22,15 +22,15 @@ git config --global user.name "Yaydoc Bot"
 git config --global user.email "noreply+bot@example.com"
 
 GIT_SSH_URL=git@github.com:$USERNAME/$REPONAME.git
-git clone --quiet $GIT_SSH_URL gh-pages
+git clone --quiet $GIT_SSH_URL gh_pages
```
Comment (imujjwal96) [inline ghpages_deploy.sh]: why this change?

Comment (pri22296) [inline ghpages_deploy.sh]: Got an error while doing `rm -rf gh-pages` probably due to `-` in the name. Considering It's only a temporary folder, I renamed it.
